quisp/utils/DefaultComponentProviderStrategy.cc

```diff
@@ -0,0 +1,29 @@
+#include "DefaultComponentProviderStrategy.h"
+
+namespace quisp {
+namespace utils {
+
+DefaultComponentProviderStrategy::DefaultComponentProviderStrategy(cModule *_self) : self(_self) {}
+
+cModule *DefaultComponentProviderStrategy::getQNode() {
+  cModule *currentModule = self->getParentModule();
```

Comment:
semantic wise this is a little weird, so this will not be called in QNode so we always start from the parent?

Reply:
actually, just I brought this code from the current code.
In the current code, there's no c++ code for QNode itself, so it works. 
But in the future, we may need to write c++ code for QNode. 
Then we will fix it, it's ok.